src/ucp/api/ucp.h

```diff
@@ -409,6 +425,33 @@ typedef enum {
 } ucp_stream_recv_flags_t;
 
 
+/**
+ * @ingroup UCP_COMM
+ * @brief flags to indicate types of events on a stream associated endpoint.
+ *
+ * This enumeration defines types of events on a stream associated endpoint.
+ * These flags will be set in the field @ref ucp_stream_poll_ep_t::flags of out
+ * parameter @a poll_eps returned by @ref ucp_stream_worker_poll routine.
+ */
+typedef enum {
+    /**
+     * There is data to receive.
+     */
+    UCP_STREAM_POLL_FLAG_IN     = UCS_BIT(0),
+
+    /**
+     * Remote peer closed the stream connection by @ref ucp_ep_close_nb in @ref
+     * UCP_EP_CLOSE_MODE_STREAM_CLOSE mode. If the flag set together
+     * with @ref UCP_STREAM_POLL_FLAG_IN, the data can be received by @ref
+     * ucp_stream_recv_nb or @ref ucp_stream_recv_data_nb before closing the
+     * local endpoint by @ref ucp_ep_close_nb. The local endpoint should be
+     * closed in @ref UCP_EP_CLOSE_MODE_STREAM_CLOSE mode too, excepting remote
+     * failure cases where @ref UCP_EP_CLOSE_MODE_FORCE may be used.
```

Comment:
More suggested wording changes:

The remote peer closed the stream connection by `@ref` ucp_ep_close_nb in `@ref` UCP_EP_CLOSE_MODE_STREAM_CLOSE mode. If `@ref` UCP_STREAM_POLL_FLAG_IN is also set, the associated data can be received by calling either `@ref` ucp_stream_recv_nb or `@ref` ucp_stream_recv_data_nb before closing the local endpoint by `@ref` ucp_ep_close_nb. The local endpoint should be closed in `@ref` UCP_EP_CLOSE_MODE_STREAM_CLOSE mode too, except in remote failure cases where `@ref` UCP_EP_CLOSE_MODE_FORCE may be used.